subprojects/core-api/src/main/java/org/gradle/plugin/use/PluginDependenciesSpec.java

```diff
@@ -117,4 +122,20 @@
      */
     PluginDependencySpec id(String id);
 
+    /**
+     * Specify a path or URL to a script plugin.
+     *
+     * <pre>
+     * plugins {
+     *     script "path/to/script-plugin.gradle"
+     *     script "http://localhost/remote-script-plugin.gradle"
+     * }
+     * </pre>
+     *
+     * @param script the path, relative to the requesting script, or URL
+     * @return this
+     * @since 4.2
+     */
+    ScriptPluginDependencySpec script(String script);
```

Comment:
See above.

Reply:
Adding an overload accepting a `URI` would allow for Kotlin build scripts to do:
```kotlin
plugins {
  script(URI.create("some/file.gradle.kts"))
}
```

To support the same thing in Groovy we would have to enhance our ad-hoc interpreter for the `plugins` DSL to also support that overload. Otherwise Groovy users would be confused when reading the `PluginDependenciesSpec` DSL documentation.

Reply:
Given `URI.create(..)` or anything creating a `URI` instance would be user code, the actual Groovy infrastructure makes it quite difficult (undesirable?) to support this.

My take is to **not** add a `URI` accepting overload to `PluginDependenciesSpec` for now.

/cc @adammurdoch, @bamboo